File: 53-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int sum = nums[0];
        int result  = sum;
        for(int i =1;i<n;i++)
        {
            if(nums[i] + sum >=0)
            {
                sum += nums[i];
                result = max(sum,  result);

            }
            else
            {
                result = max(sum,  result);
                sum = 0;
            }
        }
        
        return result;
    }
};

File: 53-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)

        sum = 0
        count =0
        result = nums[0]
        for num  in nums :
            if num + count >num :
                count += num
                result = max(result , count) 
                 
            else:
                count = num
                result = max(result , count) 
                 
        return result         
        